### PR TITLE
[Gtk] Fix leak on SizeRequest

### PR DESCRIPTION
**Pull request by plprevost**
Fixing leak on sizeRequest function by taking Requisition property from Gtk.Widget class.

### CODE REVIEW

**Comment by plprevost** [comment]
As discussed in the issue #259, this code fixes leak on SizeRequest function

File: gtk/Widget.custom [hunk 1]
```diff
@@ -393,3 +393,14 @@ public void ModifyText (Gtk.StateType state)
...
 	gtk_widget_modify_text_ptr (Handle, (int) state, IntPtr.Zero);
 }
+
+[DllImport("libgtk-win32-2.0-0.dll", CallingConvention = CallingConvention.Cdecl)]
+static extern void gtk_widget_size_request(IntPtr raw, ref Gtk.Requisition requisition);
+
+public Gtk.Requisition SizeRequest()
+{
+        Gtk.Application.AssertMainThread();
+        Gtk.Requisition requisition = default(Requisition);
```
Comment (Therzok) [inline gtk/Widget.custom]: `default(Requisition)` is basically the same thing as `new Requisition()`. Not sure how that fixes anything.

File: gtk/Widget.custom [hunk 2]
```diff
@@ -393,3 +393,14 @@ public void ModifyText (Gtk.StateType state)
...
 	gtk_widget_modify_text_ptr (Handle, (int) state, IntPtr.Zero);
 }
+
+[DllImport("libgtk-win32-2.0-0.dll", CallingConvention = CallingConvention.Cdecl)]
+static extern void gtk_widget_size_request(IntPtr raw, ref Gtk.Requisition requisition);
+
+public Gtk.Requisition SizeRequest()
+{
+        Gtk.Application.AssertMainThread();
+        Gtk.Requisition requisition = Requisition;
```
Comment (Therzok) [inline gtk/Widget.custom]: So, due to the nature of how marshalling works, we will never hit [this](https://github.com/bratsche/gtk/blob/xamarin-mac/gtk/gtkwidget.c#L4004-L4007). It seems native does not like sending over the actual widget's requisition.

**Comment by Therzok** [comment]
Still don't see how this fixes things

File: gtk/Widget.custom (hunk 2 above)
Comment (plprevost) [reply gtk/Widget.custom:403]: Hello, this fix solved our leak issue. Please, is it possible to merge this PR?
regards,

Comment (plprevost) [reply gtk/Widget.custom:403]: Hello Therzok,
The solution is to replace the instanciation of Gtk.Requisition by the reference to the property Requisition stored in the class Gtk.Widget.
So basically, the correction is the following:

`public Gtk.Requisition SizeRequest()
{
        Gtk.Application.AssertMainThread();
        Gtk.Requisition requisition = Requisition;
        gtk_widget_size_request(Handle, ref requisition);
        return requisition;
}
`

**Review by Therzok — APPROVED** [review]
Thanks for the PR!

**Pull request merged**